app/Models/Applicant.php

```diff
@@ -170,4 +172,69 @@ public function experiences_community() //phpcs:ignore
         return $this->morphMany(\App\Models\ExperienceCommunity::class, 'experienceable')
             ->orderBy('end_date', 'desc');
     }
+
+    /**
+     * Returns a Laravel QueryBuilder object which will retrieve all ExperienceSkills
+     * which are linked to this Applicant, through this Applicant's Experiences.
+     *
+     * It returns the query builder object instead of the results of the query, so that additional
+     * clauses can be added by other code.
+     *
+     * @return \Illuminate\Database\Query\Builder
+     */
+    public function experienceSkillsQuery()
+    {
+        $applicantId = $this->id;
+        return ExperienceSkill::where(function ($query) use ($applicantId) {
+            // A single where clause wraps 5 OR WHERE clauses.
+            // Each WHERE clause gets all the ExperienceSkills linked to a particular Experience type (and this Applicant).
+
+            $query->orWhere(function ($query) use ($applicantId) {
+                $query->whereIn('id', function ($query) use ($applicantId) {
+                    $query->select('experience_skills.id')->from('experience_skills')->join('experiences_work', function ($join) use ($applicantId) {
+                        $join->on('experience_skills.experience_id', '=', 'experiences_work.id')
+                            ->where('experience_skills.experience_type', 'experience_work')
+                            ->where('experiences_work.experienceable_type', 'applicant')
+                            ->where('experiences_work.experienceable_id', $applicantId);
+                    });
+                });
```

Comment:
These all look pretty similar, other then the experience types. Maybe we should abstract this out into a function. Although, I'm trying now and it doesn't seem to be cooperating.  😅

Reply:
This is what I have so far.. However, the tests assertSoftDeleted test fails.
```php 

public function experienceSkillsQuery()
    {
        $applicantId = $this->id;
        $getExperienceSkills = function ($query, $experiences, $experienceType, $applicantId): void {
            $query->whereIn('id', function ($query) use ($experiences, $experienceType, $applicantId): void {
                $query->select('experience_skills.id')
                    ->from('experience_skills')
                    ->join($experiences, function ($join) use ($experiences, $experienceType, $applicantId): void {
                        $join->on('experience_skills.experience_id', '=', "{$experiences}.id")
                        ->where('experience_skills.experience_type', $experienceType)
                        ->where("{$experiences}.experienceable_type", 'applicant')
                        ->where("{$experiences}.experienceable_id", $applicantId);
                    });
            });
        };

        return ExperienceSkill::where(function ($query) use ($getExperienceSkills, $applicantId): void {
            // A single where clause wraps 5 OR WHERE clauses.
            // Each WHERE clause gets all the ExperienceSkills linked to a particular Experience type (and this Applicant).

            $query
                ->orWhere($getExperienceSkills($query, 'experiences_work', 'experience_work', $applicantId))
                ->orWhere($getExperienceSkills($query, 'experiences_personal', 'experience_personal', $applicantId))
                ->orWhere($getExperienceSkills($query, 'experiences_education', 'experience_education', $applicantId))
                ->orWhere($getExperienceSkills($query, 'experiences_award', 'experience_award', $applicantId))
                ->orWhere($getExperienceSkills($query, 'experiences_community', 'experiences_community', $applicantId));
        });
    }

```

Reply:
Great suggestion! The trick was to write a function that returns a function. If you like my solution, merge it in!